readinessProbe/readiness/server_test.go

```diff
@@ -12,67 +11,102 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-var _ = Describe("ManagedOCS Readiness Probe", func() {
-	// Define utility constants for object names and testing timeouts/durations and intervals.
-	const (
-		timeout  = time.Second * 10
-		duration = time.Second * 10
-		interval = time.Millisecond * 250
-	)
-
-	When("managedocs reports its components as \"not ready\"", func() {
-		BeforeEach(func() {
-			ctx := context.Background()
-
-			managedOCS := &v1.ManagedOCS{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      ManagedOCSName,
-					Namespace: TestNamespace,
-				},
-				Status: v1.ManagedOCSStatus{
+var _ = Describe("ManagedOCS readiness probe behavior", func() {
+	ctx := context.Background()
+
+	managedOCS := &v1.ManagedOCS{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ManagedOCSName,
+			Namespace: TestNamespace,
+		},
+	}
+
+	Context("Readiness Probe", func() {
+		When("the managedocs resource lists its StorageCluster as not \"ready\"", func() {
+			It("should cause the readiness probe to return StatusServiceUnavailable", func() {
+				Expect(k8sClient.Get(ctx, utils.GetResourceKey(managedOCS), managedOCS)).Should(Succeed())
+
+				managedOCS.Status = v1.ManagedOCSStatus{
 					Components: v1.ComponentStatusMap{
 						StorageCluster: v1.ComponentStatus{
 							State: v1.ComponentPending,
 						},
 					},
```

Comment:
In order to validate that the readiness(or unreadiness)  is affected solely because of the specific component (which is what is tested here), I will also set all other components to Ready.

Saying that I will suggest a util function (declared in this file) that just takes the Tree statues and returns the `managedOCS` object, or alternatively a template mechanism (as we use in the tests for the reconcile loop). 

Without it, we might forget in he future to set some component status when we will introduce new components